pkgs/servers/web-apps/photoprism/default.nix

```diff
@@ -0,0 +1,127 @@
+{ lib
+, pkgs
+, stdenv
+, buildGoModule
+, fetchzip
+, fetchFromGitHub
+, coreutils
+, darktable
+, rawtherapee
+, ffmpeg
+, libheif
+, exiftool
+, nodejs
+, libtensorflow-bin
+}:
+
+let
+  version = "2020-09-21";
+  pname = "photoprism";
+
+  src = fetchFromGitHub {
+    owner = "photoprism";
+    repo = pname;
+    rev = "d030a602d47cf7279dddc301d74156ca9c21e7ef";
+    sha256 = "0gmvl4svcvdqb0y7chsn3dd13iznq4v3ms1rfvqcbrkhpqn3qhpd";
+  };
+
+  fetchModel = {name, sha256}:
+  fetchzip {
+    inherit sha256;
+    url = "https://dl.photoprism.org/tensorflow/${name}.zip";
+    stripRoot = false;
+  };
+
+  nasnet = fetchModel {
+    name = "nasnet";
+    sha256 = "0qb1sidgpx3a4synl6s4x993clgx8iv1khigbf92fbwrz66bjpbc";
+  };
+
+  nsfw = fetchModel {
+    name = "nsfw";
+    sha256 = "1y4mh7lsjw4syrgfpn6iy4qfqf6a7yi66m7j2lxyc70sd1rcfbyg";
+  };
+
+  backend = buildGoModule rec {
+    inherit pname version src;
+    doCheck = false;
+    buildInputs = [
+      coreutils
+      libtensorflow-bin
+    ];
+    postPatch = ''
+      substituteInPlace internal/commands/passwd.go --replace '/bin/stty' "${coreutils}/bin/stty"
+    '';
+    vendorSha256 = "12g0fywzqpvnmmz4rkzanvbg2m0aj5qa3q30dwczpqwbyarsg0yl";
+  };
+
+  nodeDependencies = (import ./node-composition.nix {
+    inherit pkgs nodejs;
+    inherit (stdenv.hostPlatform) system;
+  }).nodeDependencies;
+
+  frontend = stdenv.mkDerivation {
+    name = "photoprism-frontend";
+    inherit src;
+    buildInputs = [ nodejs ];
+
+    buildPhase = ''
+      runHook preBuild
+
+      pushd frontend
+      ln -s ${nodeDependencies}/lib/node_modules ./node_modules
+      export PATH="${nodeDependencies}/bin:$PATH"
+      NODE_ENV=production npm run build
+      popd
+
+      runHook postBuild
+    '';
+
+    installPhase = ''
+      runHook preInstall
+
+      mkdir $out
+      cp -r assets $out/
+
+      runHook postInstall
+    '';
+  };
+
+in
+stdenv.mkDerivation {
+
+  inherit pname version;
+
+  buildInputs = [
```

Comment:
These are only needed at runtime and can be specified with a commandline flag, environment variable, or config entry.
We probably don't need them here then?

Reply:
Yep. I don't think they would have any effect here.

One alternative would be to patch the default path in the source to explicitly refer to the path to the tool from nixpkgs. E.g. ffmpeg here: https://github.com/photoprism/photoprism/blob/46b9239026bae49a3ae2dbb57b0a374750d6f9e2/internal/config/filenames.go#L241

Currently, adding video files fails if I don't have `ffmpeg` in my PATH or pass it in the commandline / env vars.

Reply:
Any opinion on what would be cleaner?

- patching the source
- creating a wrapper that passes the commandline args

Reply:
@danielfullmer @andir @SuperSandro2000 any opinions?

Reply:
I think creating a wrapper is maybe nicer.

Reply:
if there's a commandline flag for this in photoprism, might as well use it, feels cleaner indeed